Library/Homebrew/dependency.rb

```diff
@@ -130,6 +139,8 @@ def expand(dependent, deps = dependent.deps, cache_key: nil, &block)
 
     def action(dependent, dep, &block)
       catch(:action) do
+        prune if dep.unavailable_core_formula?
```

Comment:
What's this doing? Wasn't obvious to me.

Reply:
I'll add a comment. Essentially, ignore dependencies (and their recusrsive dependencies) that don't exist because homebrew/core is untapped.

For example, if this is the dependency structure of `foo` (where `bar` is a build only dependency):

```
$ brew deps --tree --include-build foo
foo
├── bar
│   └── baz
└── formula
```

Normally (with homebrew/coe tapped), here's what `Dependency.expand Formula["foo"]` returns:

```irb
[1] brew(main)> Dependency.expand Formula["foo"]
=> [#<Dependency: "baz" [:build]>,
 #<Dependency: "bar" [:build]>,
 #<Dependency: "formula" []>]
```

Now, with homebrew/core untapped (and `bar` and `baz` not installed):

```irb
[1] brew(main)> Dependency.expand Formula["foo"]
=> [#<Dependency: "formula" []>]
```

Functionally, this is okay because build dependencies aren't needing when installing from a bottle. Also, if `#to_formula` is ever called on one of the missing deps, it will still raise a `CoreTapFormulaUnavailableError`. These are only missing from `Dependency::expand`.